### PR TITLE
Move version switcher next to the logo

### PR DESCRIPTION
**Pull request by timhoffm**
This is in preparation for adding the search field to the navbar (https://github.com/matplotlib/mpl-brochure-site/issues/50). IMHO
placing the version next to the logo also rhymes better.

Styling of the switcher is more muted (not the strong blue). It's colors
are taken from the panels on the index page.

New:
grafik

Old:
grafik

### CODE REVIEW

**Comment by timhoffm** [comment]
https://output.circle-artifacts.com/output/job/fb1c34e8-73c4-4237-b610-fd4efb28348b/artifacts/0/doc/build/html/index.html

**Comment by timhoffm** [comment]
https://output.circle-artifacts.com/output/job/da4ddcb5-9554-42b1-83f0-c1e6b8a0c7cf/artifacts/0/doc/build/html/index.html

**Comment by jklymak** [comment]
I am not a fan of it next to the logo - it's a pretty junky widget and I prefer the whitespace around the logo.

**Comment by timhoffm** [comment]
Under the logo is only because the devdocs full version is so long. In a release this should already be ok. As written, I still have to figure out how to shorten the version entry. It should look something like this then:

grafik


We could also reduce the style even more (and only display the box when hovering over it):

grafik

**Comment by oscargus** [comment]
Although I like the idea of being able to switch version in the header, I do share the concerns of @jklymak . With styling I think it is a completely different thing though. Right now it just looks so big (partly because of the width, which as you say will not really be a problem except for CI builds, but also the height).

**Comment by timhoffm** [comment]
I think I have fixed the start value to show in the switcher.

File: doc/_static/mpl.css [hunk 1]
```diff
@@ -277,3 +277,22 @@ div.twocol > div {
     display: inline-block;
     margin: 0 0.5em;
 }
+
+
+/* styling for version switcher */
+.navbar-brand {
```
Comment (jklymak) [inline doc/_static/mpl.css:283]: I like this styling, but still opposed to moving next to the logo for aesthetic reasons.

**Comment by greglucas** [comment]
I probably have a slight preference for it on the right-hand side as well, but can't say it makes a huge difference to me. Note that Pandas and Numpy have the version switcher on the right, but the Python docs have it on the left, so it is somewhat project dependent.

**Comment by timhoffm** [comment]
Note that the suggestion to move the version switcher next to the logo has gotten much appreciation with pydata-sphinx-theme https://github.com/pydata/pydata-sphinx-theme/issues/680. And they are making it the default position https://github.com/pydata/pydata-sphinx-theme/pull/698.

I haven't seen a version switcher in NumPy. But for pandas I suspect that they have just taken over the default from pydata-sphinx-theme (not least because they use the default color, which is a bit off for the general theme; and that may be a hint that they haven't yet taken the time for any customization/optimization).

**Comment by timhoffm** [comment]
This got out of date. Since pydata-sphinx-theme has the version next to the logo by default. Implementation would now be simply to revert to the theme defaults. While I still think this would be better, I'm not going to follow up due to above opposition.

**Pull request closed without merging**